### DISCUSSION TITLE
Azure B2C custom policy with Azure Active Directory support

### DISCUSSION THREAD

**Discussion by giscus** [CamiloTerevinto/Blog#10] [Announcements]
# Azure B2C custom policy with Azure Active Directory support

Implementing an IEF policy that supports local and AAD accounts with proper code documentation

https://www.camiloterevinto.com/post/azure-b2c-with-azure-active-directory

**Comment by astewartgdn** [comment]
Please check your code package.  References  Facebook in Base . Reference to  "emailVerificationControl". - probably from the display control base.  LocalSignIn User journey starts at step 2.

**Reply by CamiloTerevinto** [reply]
Hey @astewartgdn, thanks for pointing that out. I missed the email for your comment for some reason. I'll fix this over the weekend and let you know

**Reply by CamiloTerevinto** [reply]
Hey @astewartgdn, I've pushed a fix for the problems you mentioned. Thanks again!

**Reply by astewartgdn** [reply]
Still not working. The AAD Subjourney and Passwordreset SubJourney are ok 
Validation failed: 
2 validation error(s) found in policy "B2C_1A_TRUSTFRAMEWORKEXTENSIONS" of tenant "xxx.onmicrosoft.com".
Orchestration step order "2" in user journey "xxx_SignUpSignIn" in policy "B2C_1A_TrustFrameworkExtensions" of tenant "xxx.onmicrosoft.com" is followed by a claims provider selection step and must be a claims exchange, but it is of type "InvokeSubJourney"
User journey "xxx_SignUpSignIn" in policy "B2C_1A_TrustFrameworkExtensions" of tenant "xxx.onmicrosoft.com" has step 3 with 3 claims exchanges. 
It must be preceded by a claims provider selection in order to determine which claims exchange can be used.

**Reply by CamiloTerevinto** [reply]
Hey @astewartgdn, sorry I haven't had a lot of time lately. I've pushed a few updates to the TrustFrameworkExtensions policy and I'm able to import it now.

**Reply by astewartgdn** [reply]
Don't concern yourself with fix timeline. I just think yours is one of the best reference builds for Custom policies and I adopted several of your points in my main code, so I would like it clean for newcomers. 

Good update, deployed SignUpOrSignIn.xml and PasswordReset.xml and tested all flows with local account and AAD account, all working. Sent a pull for the typo.

ProfileEdit.xml fail to load 
1 validation error(s) found in policy "B2C 1A PROFILEEDIT" of tenant "xxx.onmicrosoft.com". Claim type "alternativeSecurityld" is the input claim of technical profile "AAD-UserReadUsinqAlternativeSecurityld" in step "3" of user journey "ProfileEdit" but it is not an output claim in any of the previous steps.

**Reply by CamiloTerevinto** [reply]
Hey @astewartgdn, that's fixed now, I was able to import all policies successfully with the last commit I just pushed. Thanks for the help in testing all this!

If you want to further improve/extend this sample, I'd be glad to accept PRs as well :)
